tests/collections/asr/mixins/adapters/test_asr_adapter_mixin.py

```diff
@@ -467,6 +603,74 @@ def test_squeezeformer_forward_mha(self, squeezeformer_ctc_adapter, name):
 
         assert torch.mean(torch.abs(origial_output - new_output)) < 1e-5
 
+    @pytest.mark.unit
+    @pytest.mark.parametrize('adapter_type', ['linear', 'attn'])
+    @pytest.mark.parametrize(
+        'name', ['adapter_0', 'encoder:adapter_0', 'transf_encoder:adapter_0', 'transf_decoder:adapter_0']
+    )
+    def test_canary_forward_mha(self, multitask_model, name, adapter_type):
+        multitask_model.eval()
+        torch.random.manual_seed(0)
+        input_signal = torch.randn(2, 512)
+        input_signal_length = torch.tensor([512, 512], dtype=torch.int32)
+        transcript = torch.randint(0, multitask_model.tokenizer.vocab_size, size=(2, 10))
+        transcript_len = torch.tensor([10, 9], dtype=torch.int32)
+
+        origial_output = multitask_model(
+            input_signal=input_signal,
+            input_signal_length=input_signal_length,
+            transcript=transcript,
+            transcript_length=transcript_len,
+        )
+        og_logprob = origial_output[0]
+        og_enc_out = origial_output[2]
+
+        if adapter_type == 'attn':
+            adapter_type = 'transf_mha' if 'transf' in name else 'mha'
+
+        multitask_model.add_adapter(name=name, cfg=get_adapter_cfg(in_features=128, atype=adapter_type, proj_dim=4))
```

Comment:
If the name is '' (default), it will use whatever the class refers to as default - i generally made it use encoder as default because every model will have an encoder in asr usually.